### ISSUE TITLE
Ensure all Buildbot builders listed for servo repo in Homu configuration are valid

### ISSUE THREAD

**Issue by aneeshusa** [servo/saltfs#443]
The Homu configuration (`homu/cfg.toml`) lists a number of Buildbot repos that are used for gating/try builds, in the `builders` and `try_builders` fields of `repo.servo.buildbot`. Recently, the `windows` builder [got renamed](https://github.com/servo/saltfs/commit/ac587293db495213cf9fbf16881206f5cb42e88a) to `windows-dev` in the Buildbot configuration without updating the Homu configuration, causing breakage. We should ensure that all the builders listed are valid builders.

Files:
- Make a new test file `tests/sls/homu/valid_builders.py`.

The test should read in the templated, deployed Homu configuration using the `toml` library, and examine those two fields, checking them against the list of builders specified in the deployed Buildbot configuration to make sure they all exist.

Note: successfully reading the deployed Buildbot configuration is not that easy, because we need to import the `master.cfg` file, but it doesn't have a `.py` extension! If you look at how Buildbot 8 [does this](https://github.com/buildbot/buildbot/blob/2e5c5ced05615ec80f03b40d7bfe80322cf04f9f/master/buildbot/config.py#L121), it looks like this requires usage of `exec`. (I don't see anything in either `importlib` or `imp` that would help here.) I consider this a necessary evil for now.

Note 2: there may be problems due to the tests using Python 3 and the Buildbot configuration using Python 2; it is likely that a Python 2 subprocess will be needed to extract the list of builders from the configuration.

Tip: if you rename the `master.cfg` file to `master.py`, you can import the module from a Python REPL to look at the available bindings to get to the list of builders.

**Comment by loganmhb** [comment]
I am interested in giving this a try.

**Comment by aneeshusa** [comment]
@loganmhb Great! Let us know if you have any questions.

**Comment by loganmhb** [comment]
Just starting out with this, but it looks like the imp.load_source() function might enable us to load master.cfg without the use of exec, if that seems preferable. (See http://stackoverflow.com/questions/2601047/import-a-python-module-without-the-py-extension)

**Comment by aneeshusa** [comment]
@loganmhb If you can make it work without `exec`, I'd be overjoyed. Thanks for double-checking the docs!

**Comment by aneeshusa** [comment]
@loganmhb How's this going? Do you have any more questions?

**Comment by loganmhb** [comment]
@aneeshusa apologies for slow progress here -- it's taken me a little while to get up to speed with the salt build process. I've got the local test environment with Vagrant running, but I can't figure out where to find the deployed Buildbot configuration -- can you point me in the right direction for that? Thanks!

**Comment by aneeshusa** [comment]
@loganmhb There's a bit of indirection in the [`Salt code that deploys the Buildbot master configuration`](https://github.com/servo/saltfs/blob/654e20615a1b7220ab3983bdda2035663976a220/buildbot/master/init.sls#L22), but you should be able to find it in `/home/servo/buildbot/master/`. I'll also update the issue description, sorry I didn't include that originally!

**Comment by loganmhb** [comment]
Hmm, there must be something wrong with how I'm doing the setup then -- the /home/servo directory is empty. I'll try to figure out where that's going wrong.

**Comment by aneeshusa** [comment]
@loganmhb You need to be using the `servo-master1` Vagrant VM - did you use another one perchance?

**Comment by jdm** [comment]
@loganmhb Did you sort out the issue? Are you still working on this?

**Comment by loganmhb** [comment]
So sorry for disappearing! Thought I had already responded to this but I
didn't send it apparently.

I haven't been able to work out my virtualenv/environment issues yet - have
had limited time to work on it. I may be able to sometime soon, but I don't
want to block progress if someone else wants to handle it, so if you want
to reassign you should go ahead.

**Comment by aneeshusa** [comment]
#562 is another place this would have been helpful! (It always seems to be windows-related...)

**Comment by charlesvdv** [comment]
I'm interrested to work on this but I can get the vagrant VM to run... 
I got this when running ```vagrant up servo-master1 --provision```:
```
[INFO    ] Loading fresh modules for state activity
[ERROR   ] No contents found in top file
local:
----------
          ID: states
    Function: no.None
      Result: False
     Comment: No Top file or external nodes data matches found.
     Started: 
    Duration: 
     Changes:   
```
Should I make something more ?

**Comment by aneeshusa** [comment]
@charlesvdv go for it!

**Comment by aneeshusa** [comment]
Fixed by #571.
